### DISCUSSION TITLE
Blocking BUG - Client Browser Flow is not continuing after User log in from external Identity Provider

### DISCUSSION THREAD

**Discussion by cgeorgilakis** [keycloak/keycloak#12893] [Ideas]
Keycloak provides full functionality for using it both as Service Provider and Identity Brokering.
However, a BLOCKING BUG exists for using Keycloak as Identity Brokering and Social Login. Our users are mainly from external Identity Providers and we have many Clients with different functionality ( some of them use OTP compulsorily, others being able to deny access, multi factor authentication etc).

The problem is that any flow/ execution of  Client Browser Flow is not executed after User log from external Identity Provider( case of Username Password Form and Identity Provider Redirector). For Keycloak Users everything is ok. An example can be found in [this jira issue](https://github.com/keycloak/keycloak/issues/10250) .

We want to mention that configure Identity Provider "Post Login Flow" is not acceptable for our case. Our flows ( OTP, deny access, multi factor authentication) is based on Client Browser Flow not in  Identity Provider "Post Login Flow". Client Browser Flow MUST BE CONTINUED after Identity Provider "First Login Flow" and "Post Login Flow" have finished.

Our implementation saves in AuthNotes of AuthenticationSessionModel all needed information for Client Browser Flow ( AuthenticationProcessor.CURRENT_AUTHENTICATION_EXECUTION of Client Browser Flow, flowid) before redirecting to Identity Provider login. After Identity Provider login has successfully finished, Keycloak can retrieve all needed information for Client Browser Flow from AuthNotes of AuthenticationSessionModel and proceed with Client Browser Flow. Basic Identity Provider login flows have been tested together with Client Browser flows (OTP,  deny access, multi factor authentication).

We have made a [PR](https://github.com/keycloak/keycloak/pull/12891) for this. Could you review it? Could you help us to overcome little tests failures? Details can be found in the [PR](https://github.com/keycloak/keycloak/pull/12891).

**Comment by bonnm** [comment]
This bug forced me to have all my authentication executions beeing defined twice. Once in the browser flow to get them executed in case of a cookie auth, and once in a post broker flow to get them executed in case of a brokered login. Would be happy if this bug gets resolved...
